packages/fx-core/src/plugins/resource/bot/enums/hostType.ts

```diff
@@ -0,0 +1,7 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+export enum HostType {
```

Comment:
From my understanding the blog you attached mostly said that "numeric enum" is bad because it breaks type safety of Typescript, but this "string enum" is okay.
![image](https://user-images.githubusercontent.com/9698542/157368718-ce48de57-d547-4d4a-8b94-a62109305f74.png)

Reply:
As we discussed offline, I have changed the new `HostType` into literal types. There are still some enum type in bot plugin. @Siglud will refactor the bot plugin later.